app/src/main/java/com/eis/geoCalendar/network/SMS/SMSStringEventNetworkManagerAdapter.java

```diff
@@ -0,0 +1,103 @@
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.network.NetworkManager;
+import com.eis.communication.network.listeners.GetResourceListener;
+import com.eis.communication.network.listeners.InviteListener;
+import com.eis.communication.network.listeners.RemoveResourceListener;
+import com.eis.communication.network.listeners.SetResourceListener;
+import com.eis.geoCalendar.gps.GPSPosition;
+import com.eis.geoCalendar.network.SMS.ListenerAdapters.GetResourceListenerAdapter;
+import com.eis.geoCalendar.network.SMS.ListenerAdapters.RemoveResourceListenerAdapter;
+import com.eis.geoCalendar.network.SMS.ListenerAdapters.SetResourceListenerAdapter;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smsnetwork.SMSFailReason;
+import com.eis.smsnetwork.SMSNetworkManager;
+
+import java.util.ArrayList;
+
+/**
+ * Simple network manager Adapter for {@link SMSStringEvent}s that uses {@link com.eis.smsnetwork.SMSNetworkManager}
+ * as real network manager. Uses Singleton Design pattern since {@link com.eis.smsnetwork.SMSNetworkManager} is a singleton
+ * <p>
+ * This class was created because of conflicts with the defined interfaces and the available network dictionary's interfaces
+ *
+ * @author Turcato
+ */
+public class SMSStringEventNetworkManagerAdapter implements NetworkManager<GPSPosition, ArrayList<SMSStringEvent>, SMSPeer, SMSFailReason> {
+    private static SMSStringEventNetworkManagerAdapter instance;
+
+    private SMSNetworkManager smsNetworkManager;
+    private SMSNetworkEventUser myself;
+
+    /**
+     * Private constructor since this class uses Singleton DP
+     *
+     * @param appContext The application's context
+     * @param myself     The user of the current Device
+     */
+    private SMSStringEventNetworkManagerAdapter(@NonNull Context appContext, @NonNull SMSNetworkEventUser myself) {
+        smsNetworkManager = new SMSNetworkManager();
+        smsNetworkManager.setup(appContext.getApplicationContext());
+        this.myself = myself;
+    }
+
+    /**
+     * @param appContext The application's context
+     * @param myself     The user of the current Device
+     * @return The only instance the device is allowed to access
+     */
+    public static SMSStringEventNetworkManagerAdapter getInstance(@NonNull Context appContext, @NonNull SMSNetworkEventUser myself) {
+        if (instance == null)
+            return instance = new SMSStringEventNetworkManagerAdapter(appContext, myself);
+        else
+            return instance;
+    }
+
+    /**
+     * @param key                 The key of the Resource (event) to add
+     * @param value               A list of resources (events), identified by the given key, to add
+     * @param setResourceListener A listener that waits for the resource to be added
+     */
+    @Override
+    public void setResource(GPSPosition key, ArrayList<SMSStringEvent> value, SetResourceListener<GPSPosition, ArrayList<SMSStringEvent>, SMSFailReason> setResourceListener) {
```

Comment:
Why do we use only one key to identify a list of events? Do you mean a list of events created by different users, but in the same position?

Reply:
Can't really answer that since I haven't understood it myself, I have the same doubt, but I had to implement Crema's interface so it's there😂

Reply:
Oh perfect😂 I am also thinking about the database: we'll have a list of events identified by the same key (as, if I'm not wrong, the events in db are identified by their position, right?), might this cause errors when saving them? Or maybe we'll save only one record, where the key is the GPSPosition and the value in a list of events?

Reply:
If I'm not mistaken, the db we have now saves events as a pair GPSPosition:String, so One record for each event and they both form the key.
So the db should allow saving events at the same location if the content is different.
But I'm not sure how the rest of the code could handle "collisions" (since I'm not very handy with it)

Reply:
Ah ok, I'm sorry, I lost this detail about the db. So, theoretically, there won't be problems when saving events. 
Me too, I'm not totally convinced about this choice (I mean having a list of events with the same key). 
Maybe we should ask @CremaLuca, as he wrote the interface.

Reply:
No need to be sorry, probably I should have been more suspicious myself

Reply:
Don't worry, you're making a lot of effort for this app, you can't be suspicious about everything!
You're doing a great job.

Reply:
The database currently handles collisions matching the whole events, with all their parameters since that was the request I got. I can change this into "equals returns true --> collision" but it will take most of tomorrow

Reply:
No need to change the db, I think.

Reply:
> Don't worry, you're making a lot of effort for this app, you can't be suspicious about everything!
> You're doing a great job.

Ma cosí mi commuovo🙈

Reply:
> The database currently handles collisions matching the whole events, with all their parameters since that was the request I got. I can change this into "equals returns true --> collision" but it will take most of tomorrow

It still remains the best choice in my opinion
Also, go to sleep

Reply:
![SAVE_20200227_012456](https://user-images.githubusercontent.com/56487105/75400588-ffd0cc00-58ff-11ea-86b5-b78be9a0c407.jpg)

Reply:
Jokes aside I'm kinda having a bruh moment for various reasons, I'll go to sleep in a moment dw

Reply:
😂good boi